ethereum/contracts/Getters.sol

```diff
@@ -27,9 +27,16 @@ contract Getters is State {
     }
 
     function chainId() public view returns (uint16) {
+        if (evmChainId() != block.chainid) {
+            return type(uint16).max;
```

Comment:
This makes all upgrades compatible between all forks. Do we want to introduce some shifting here e.g. `uint16.max -32 + block.chainid % 32` to reduce the odds of collisions?

Reply:
Latest approach we discussed with @evan-gray was we should differenciate between the chains using `block.chainid` when applying an ugrade.
I have no strong opinion here. This could make an upgrade on a fork easier but i think we should still have the `block.chainid` check.

Reply:
Chatted with @calebrate and @kcsongor and we fall in favor of the static return type(uint16).max; (e.g. all chains ending with 01 end up the same 🙁)